test/functional/shared.js

```diff
@@ -192,12 +192,24 @@ class EventCollector {
   }
 }
 
+function withMonitoredClient(configuration, callback) {
```

Comment:
one of the nice things about `withClient` is that it will close the client for you when its done, it effectively takes a `body` function which it runs to completion and always ensures the client is closed afterward. The goal was that we'd be able to use it as a type of mocha "middleware" but that might be ambitious:

```
it('should work', withClient((client, done) => {
   // client here is connected, otherwise done would have already been called with the connection error
  client.db().command('admin.$cmd', { ping: 1 }, err => {
    expect(err).to.not.exist;
    done();
  });
});
```
No need to even worry about closing the client, and also allows us to potentially have a factory to do fancy stuff like reusing clients in the future (if they have the same configuration). 

I wonder if we can make `withMonitoredClient` similar:
```
it('should work', withMonitoredClient(['find'], (client, events, done) => {
  ...
});
```

The thing we are abstracting is the lifetime management of the `client`, and collection of events. It would not return a `Collection`, that's just something you do in your test.